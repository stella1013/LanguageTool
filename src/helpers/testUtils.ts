//import { ConnectedRouter } from 'connected-react-router';
//import React from 'react';
//import { Provider } from 'react-redux';
////import { mergeDeepRight } from 'ramda';
//import rootReducer from '../redux/reducers';
////import  createStoreWithMiddleWare  from '../redux/store';
//import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
//import { AppState } from '../redux/store';
//import ReduxThunk  from 'redux-thunk';
//import { createStore, applyMiddleware } from 'redux';
//import rootReducer from '../redux/reducers';
//import thunk from 'redux-thunk';

/*
	* Returns Shallow Wrapper containing node(s) with the given data-test value
	* @param {Shallow Wrapper} - Enzyme shallow wrapper to search within
	* @param {string} val - Value of data-test attribute for search.
	* @returns {Shallow Wrapper}
	*/ 
	export const findByTestAttr = (wrapper:any, val:string) => {
		return wrapper.find(`[data-test="${val}"]`);
	}

	export const storeFactory = (initialState:any) =>{
		const createMockStore = configureMockStore();
		return createMockStore(initialState);
	}
	/*
	export const makeMountRender = (Component:any, defaultProps = {}) => {
		return (customProps = {}) => {
		  const props = {
			...defaultProps,
			...customProps
		  };
		  return mount(<Component {...props} />);
		};
	  };

	  export const makeStore = (customState = {}) => {
		const root = rootReducer({}, { type: '@@INIT' });
		const state = mergeDeepRight(root, customState);
		return createStoreWithMiddleWare(rootReducer, state);
	  };
	  export const snapshotify = (reactWrapper:any) => {
		return reactWrapper.html();
	  };

	  export const reduxify = (Component, props = {}, state = {}) => {
		return function reduxWrap() {
		  return (
			<Provider store={makeStore(state)}>
			  <ConnectedRouter history={history}>
				<Component {...props} />
			  </ConnectedRouter>
			</Provider>
		  );
		}
	  };

*/