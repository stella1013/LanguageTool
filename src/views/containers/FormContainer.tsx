import * as React from 'react';
// import { Row, Col } from 'react-bootstrap';
import CustomInput from './Fields/CustomInput';
import { InputField, DropDown, FileUpload, Header } from './Fields/FieldTypes';

interface FormState {
	isReady: boolean;
	validSearch: boolean;
	formValues: Array<InputField | DropDown | FileUpload | Header>;
}

class FormContainer extends React.Component<any, FormState> {
	constructor(props: any) {
		super(props);
		this.state = {
			isReady: false,
			validSearch: false,
			formValues: []
		};
	}

	render() {
		const formElementsArray = [];
		//console.log(this.props.formScaffold);
		
		if (this.props.formScaffold) {
			let key: any;
			for (key in this.props.formScaffold) {
				formElementsArray.push({
					id: key,
					config: this.props.formScaffold[key]
				});
			}
		}
		
		let form: any = null;
		if (formElementsArray.length > 0) {
			form = (
					
						formElementsArray.map((formElement:any) => {
						let elementConfig = formElement.config.elementConfig
							? formElement.config.elementConfig
							: null;
						return (
							<CustomInput
								data-test="form-fields"
								required="true"
								key={formElement.id}
								ident={formElement.id}
								id={formElement.id}
								elementType={formElement.config.elementType}
								size={formElement.config.elementSize}
								elementConfig={elementConfig}
								value={formElement.config.selectedValue.value}
								displayValue={formElement.config.selectedValue.displayValue}
								//postLabel={formElement.config.postLabel}
								changed={
									// this.inputChangedHandler
									(
										event: React.FormEvent<HTMLInputElement>,
										//id: any,
										newVal: any
									) => {
										this.props.inputChangedHandler(
											/*event, formElement.id, */ newVal
										);
									}
								}
							/>
						);
					})
				
			);
		}

		return (
			<div className="row" data-test="component-form">
				{form}
			</div>
		);
	}
}
export default FormContainer;
