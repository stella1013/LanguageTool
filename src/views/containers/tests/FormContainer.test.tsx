import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import { findByTestAttr } from '../../../helpers/testUtils';
import thunk from 'redux-thunk';
import FormContainer from '../FormContainer';



describe('Form Container', () => {
	/*
	 * Factory function to create a Shallow Wrapper for the App Component
	 * @function setup
	 * @param {object} props - component props
	 * @param {object} state - initial state
	 * @returns {Shallow Wrapper}
	 */
    const createMockStore = configureStore([thunk]);
    const initialState = {};
    const store = createMockStore(initialState);
    let wrapper: ShallowWrapper;
	
	describe('rendering', () => {
       //let wrapper: any;
		beforeEach(() => {
            wrapper = shallow(<FormContainer showModal={false}/>);
		});


		it('renders without crashing', () => {
			//const wrapper = setup();
			const formComponent = findByTestAttr(wrapper, 'component-form');
			expect(formComponent.length).toBe(1);
		});
	
		it('render matches snapshot', () => {
			//const wrapper = setup();
			expect(wrapper).toMatchSnapshot();
		});
		it('initial validSearch `state` should be false', () => {
			//const wrapper = setup();
			const validSearchState = wrapper.state('validSearch');
			expect(validSearchState).toBe(false);
		});
		/*
		it('renders form fields', () => {
			//const wrapper = setup();
			const formFields = findByTestAttr(wrapper, 'form-fields');
			expect(formFields.length).toBeGreaterThan(0);
		});
	
	
		it('initial isReady `state` should be set to true', () => {
			//const wrapper = setup();
			const initReadyState = wrapper.state('isReady');
			expect(initReadyState).toBe(true);
		});
		*/
		
	});

	describe('state updates', () => {});
});
