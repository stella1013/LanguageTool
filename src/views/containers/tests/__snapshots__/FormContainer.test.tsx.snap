// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Container rendering render matches snapshot 1`] = `ShallowWrapper {}`;
